### PR TITLE
[BREAK][ENTERPRISE] Limit presence statuses to 200 concurrent users when running monolith to keep performance

### PR DESCRIPTION
**Pull request by sampaiodiego**
[ARCH-131]
[ARCH-81]

## Issue(s)

ARCH-81

## Further comments

[ARCH-131]: https://rocketchat.atlassian.net/browse/ARCH-131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ARCH-81]: https://rocketchat.atlassian.net/browse/ARCH-81?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: ee/packages/presence/src/Presence.ts [hunk 1]
```diff
@@ -1,39 +1,73 @@
...
+import { License, ServiceClass } from '@rocket.chat/core-services';
 import { UserStatus } from '@rocket.chat/core-typings';
-import { Users, UsersSessions } from '@rocket.chat/models';
+import { Settings, Users, UsersSessions } from '@rocket.chat/models';
 
 import { processPresenceAndStatus } from './lib/processConnectionStatus';
 
+const MAX_CONNECTIONS = 200;
+
+const connsPerInstance = new Map<string, number>();
```
[the remark this answers is not in the document]
Comment (rodrigok) [reply ee/packages/presence/src/Presence.ts:11]: I didn't see any reason as well

File: apps/meteor/app/lib/server/startup/settings.ts [hunk 2]
```diff
@@ -14,6 +14,13 @@ settingsRegistry.add('Initial_Channel_Created', false, {
 	hidden: true,
 });
 
+// this settings will let clients know in case presence has been disabled
+settingsRegistry.add('Presence_broadcast_disabled', false, {
+	type: 'boolean',
+	public: true,
+	// hidden: true,
```
Comment (rodrigok) [inline apps/meteor/app/lib/server/startup/settings.ts:21]: Why the setting is not hidden anymore @ggazzo ?

Comment (ggazzo) [reply apps/meteor/app/lib/server/startup/settings.ts:21]: even being public, hidden settings are not sent to client =/

**Review by rodrigok — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Review by Nadineersej — COMMENTED** [review]
Y

**Pull request merged**